frontend/server/controllers/CourseController.php

```diff
@@ -195,60 +205,65 @@ public static function apiClone(Request $r) {
         if (OMEGAUP_LOCKDOWN) {
             throw new ForbiddenAccessException('lockdown');
         }
-        $original_course = CoursesDAO::getByAlias($r['course_alias']);
-        $offset = round($r['start_time']) - strtotime($original_course->start_time);
-        $auth_token = isset($r['auth_token']) ? $r['auth_token'] : null;
+
+        self::authenticateRequest($r);
+        self::validateCourseAlias($r);
+        self::validateClone($r);
+
+        $originalCourse = CoursesDAO::getByAlias($r['course_alias']);
+        if (is_null($originalCourse)) {
+            throw new NotFoundException('courseNotFound');
+        }
+        $offset = round($r['start_time']) - strtotime($originalCourse->start_time);
 
         DAO::transBegin();
-        $response = [];
+
         try {
             // Create the course (and group)
-            $response[] = self::apiCreate(new Request([
+            [$courseId, $aclId] = CourseController::createCourseAndGroup(new Courses([
                 'name' => $r['name'],
-                'description' => $original_course->description,
+                'description' => $originalCourse->description,
                 'alias' => $r['alias'],
-                'start_time' => $r['start_time'],
-                'finish_time' => strtotime($original_course->finish_time) + $offset,
-                'public' => 0, // All cloned courses start in private mode
-                'auth_token' => $auth_token
-            ]));
+                'school_id' => is_null($r['school']) ? null : $r['school']->school_id,
+                'start_time' => gmdate('Y-m-d H:i:s', $r['start_time']),
+                'finish_time' => gmdate('Y-m-d H:i:s', strtotime($originalCourse->finish_time) + $offset),
+                'public' => 0,
+                'show_scoreboard' =>$r['show_scoreboard'] == 'true',
```

Comment:
```suggestion
                'show_scoreboard' => $r['show_scoreboard'] == 'true',
```

también, puedes hacer `$r->ensureBool('show_scoreboard')` para quitar la comparación con `'true'`?

Reply:
¿Te refieres a `boolval($r['show_scoreboard'])`?, porque el que tu mencionas sólo sirve para validar, pero no para asignar el valor.